### PR TITLE
[ML] integrate model download into put trained model API

### PR DESCRIPTION
**Pull request by hendrikmuhs**
This PR enables downloading packaged models from `ml-models.elastic.co`, an endpoint provided by Elastic. Elastic provided models begin with a `.`, which is a private namespace that does not interfere with user models(the `.` prefix is disallowed for them). If a user puts a packaged model, the model gets automatically downloaded. For air-gapped environments it is possible to load models from a file.

earlier changes: #95175, #95207

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java [hunk 1]
```diff
@@ -256,7 +293,69 @@ protected void masterOperation(
...
+        trainedModelConfig.setTags(resolvedModelPackageConfig.getTags());
+        trainedModelConfig.setModelPackageConfig(
+            new ModelPackageConfig.Builder(resolvedModelPackageConfig).resetPackageOnlyFields().build()
+        );
+    }
+
+    private void triggerModelFetchIfNecessary(
+        String modelId,
+        ModelPackageConfig modelPackageConfig,
+        boolean waitForCompletion,
```
Comment (hendrikmuhs) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java:336]: `waitForCompletion` still misses some parts in the package loader code. I will do them in a separate PR.

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Review by droberts195 — APPROVED** [review]
LGTM if you could change the release note.

Also one non-essential suggestion about the tests.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java [hunk 2]
```diff
@@ -256,7 +293,69 @@ protected void masterOperation(
             listener::onFailure
         );
 
-        checkModelIdAgainstTags(config.getModelId(), modelIdTagCheckListener);
+        if (config.isPackagedModel()) {
+            resolvePackageConfig(config.getModelId(), ActionListener.wrap(resolvedModelPackageConfig -> {
+                try {
+                    trainedModelConfig.validateNoPackageOverrides();
+                } catch (ValidationException e) {
+                    listener.onFailure(e);
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java:301]:
```suggestion
                    listener.onFailure(e);
                    return;
```

**Comment by hendrikmuhs** [comment]
@elasticmachine update branch

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**